demo/demo.md

```diff
@@ -232,6 +232,10 @@ When using the `required` property, a custom message can be inserted using the `
   ```
 </auro-accordion>
 
+### Functionally determining error state
+
+To make this error state dynamic, use a function to determine whether or not this component is in error state by conditionally returning this message string based on your desired criteria. 
+
```

Comment:
I'd prefer that a phrase like this is added around line 176. The `error` property is clearly listed as a `string` property allowing for any string to be entered into it. How that string is generated is up to the user outside the scope of the element. 

By definition, that is dynamic. Also, the function does not determine of the element is in error, the function would determine if the data is invalid. Based on that return, the element would then have a `valid` or `invalid` state that is illustrated by entering a string into the `error` property.